### PR TITLE
Fix how BCD renderer determines the need for notes

### PR DESCRIPTION
**Pull request by Elchi3**
The BCD table renderer assumes that you only need to render notes when there are `some((item) => item.prefix || item.notes || item.alternative_name || item.flags)`. This is a wrong assumption. 
The table should also render notes whenever the support statement is complex (an array) with more than one entry. It could be that support used to be partial, or that support was removed. Also, the range in which full support was gained should be shown.

Fixes: https://github.com/mdn/yari/issues/3839

Test pages:
http://localhost:3000/en-US/docs/Web/CSS/gap#support_in_grid_layout
http://localhost:3000/en-US/docs/Web/CSS/appearance#browser_compatibility
http://localhost:3000/en-US/docs/Web/SVG/Attribute/stop-opacity#browser_compatibility
http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/@@iterator#browser_compatibility
http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSource#browser_compatibility

(and probably many many more given array support statements are not that rare in BCD)

### CODE REVIEW

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (peterbe) [inline client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:373]: Can you put some of the description you put into the PR description here as a code comment.
It stands out like a rather "odd" piece of business logic. It might make perfect sense inside the BCD project but perhaps it would be less surprising if we can write someone down here to remind the Yari developer looking at this.

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (peterbe) [inline client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:314]: The code comment doesn't match the implementation. A more accurate implementation, assuming the code comment is correct, would be something like this:
```js
isEqualSet(new Set(Object.keys(item)), new Set(['version_added', 'releaseDate']))
```

Or, perhaps it would be a more "sustainable" implementation to do something like this:

```js
Object.keys(item).includes('version_added') && Object.keys(items).includes('releaseDate')
```

**Review by peterbe — CHANGES_REQUESTED** [review]
I can't say I actually fully understand the business logic but I can at least judge the connection between the code and the code comments :)

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (Elchi3) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:378]: I think it is odd that this was named "hasNotes". It should rather be "hasMultipleSupportStatements" or so.
I added a comment.

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (Elchi3) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx]: Right, I was too lazy to actually compare anything and I thought it would be enough to rely on the length. Fixed.

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (peterbe) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:378]: I can kinda understand why it was called `hasNotes`. 
On https://developer.mozilla.org/en-US/docs/Web/CSS/appearance#browser_compatibility you can click to expand...
Screen Shot 2021-05-24 at 3 13 57 PM
...to be able to see and read about what the "-x-" thing means.

Comment (peterbe) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:378]: After all, the button has a tooltip that is "Open implementation notes". 

But if you think it's not valuable to be able to expand those that now cease to be expandable, then I trust you.

**Review by peterbe — APPROVED** [review]
I know it's a nit, but if you have the energy, would you mind changing the code comment sentence on line 313 to be a full sentence (capital first letter and full stop) and make it not longer than 80 characters?

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (foolip) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:378]: Sounds like in addition to multiple support statements, the "show more stuff" arrow should be shown if there are notes for a single entry too?

And the tooltip will now be wrong in some cases, should it depend on whether there are notes, or always be "show details" or something?

Comment (Elchi3) [reply client/src/document/ingredients/browser-compatibility-table/feature-row.tsx]: > Sounds like in addition to multiple support statements, the "show more stuff" arrow should be shown if there are notes for a single entry too? 

Oh yes, thanks, I was focusing too much on the issue at hand. The original check for hasNotes needs to stay but it was incomplete. Added this back.

> And the tooltip will now be wrong in some cases, should it depend on whether there are notes, or always be "show details" or something?

Not sure what you mean exactly but I noted the notes were a bit off, so I fixed them. The order of the icons and the corresponding notes should now align. 
Sometimes BCD provides notes that say the same thing that is already implied by data, but I guess that's more of a data issue.

**Comment by peterbe** [comment]
Alignment looks a bit odd now:
Screen Shot 2021-05-25 at 12 38 16 PM

Compared to
Screen Shot 2021-05-25 at 12 38 59 PM

**Comment by peterbe** [comment]
I suppose it tries to be 2 columns. Is that right?
If so, they're not aligned at the top.

**Comment by Elchi3** [comment]
I think this problem is on production, too. It happens whenever there are two or more columns and one text uses an icon (like prefix "-x-" here) and another does not. 
I tried to debug it but my CSS skills are too weak.

**Comment by Elchi3** [comment]
What's weird is that prefix is the only case that displays the icon "-x-" again. You don't have the flag again for flagged features, for example. So, maybe it was never meant to show the icon in the text again and it happened by accident that prefix has this.

**Comment by Elchi3** [comment]
I've applied the class `footnote` to all of the footnotes now. That way no icon is shown twice and alignment looks alright to me.

**Comment by peterbe** [comment]
Pardon my ignorance, again, but what this PR brings is that I can now expand to see when the "Partial support" first came in?
Screen Shot 2021-06-01 at 2 24 15 PM
That's something that I can't see in prod. 
Is that a good example of what this PR adds?

**Comment by peterbe** [comment]
The other thing I don't understand, with how it relates to notes, is that now which version is displayed changes. 
In prod it's the full support one. But in this branch, it's that other (alternative name??) one.

Branch:
Screen Shot 2021-06-01 at 2 27 52 PM

Prod:
Screen Shot 2021-06-01 at 2 28 11 PM

**Comment by Elchi3** [comment]
> Pardon my ignorance, again, but what this PR brings is that I can now expand to see when the "Partial support" first came in?
> That's something that I can't see in prod.
> Is that a good example of what this PR adds?

Yes, exactly. Before this PR, a lot of the history is just ignored including this partial support information :(

> The other thing I don't understand, with how it relates to notes, is that now which version is displayed changes.
In prod it's the full support one. But in this branch, it's that other (alternative name??) one.

I assume this is because this branch uses an older BCD version. I think we fixed that in the data. The BCD PR is quite fresh https://github.com/mdn/browser-compat-data/pull/10561

**Comment by peterbe** [comment]
> > Pardon my ignorance, again, but what this PR brings is that I can now expand to see when the "Partial support" first came in?
> > That's something that I can't see in prod.
> > Is that a good example of what this PR adds?
> 
> Yes, exactly. Before this PR, a lot of the history is just ignored including this partial support information :(
> 
> > The other thing I don't understand, with how it relates to notes, is that now which version is displayed changes.
> > In prod it's the full support one. But in this branch, it's that other (alternative name??) one.
> 
> I assume this is because this branch uses an older BCD version. I think we fixed that in the data. The BCD PR is quite fresh [mdn/browser-compat-data#10561](https://github.com/mdn/browser-compat-data/pull/10561) So, the ordering is a BCD fix.

If you feel like my comments (of ignorance :) ) are answered, I'm still r+ on this PR.

**Comment by peterbe** [comment]
Do you want me to merge this or did you want to wait for an approval from @foolip ?

**Comment by foolip** [comment]
I don't know this code so I wouldn't be a very good reviewer :)

**Comment by Elchi3** [comment]
@peterbe yes, I think you should merge this :) Thanks for your great review (as always!)

**Pull request merged**
